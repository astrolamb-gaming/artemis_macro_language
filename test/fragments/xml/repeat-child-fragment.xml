<mission_data version="1.0">
  <templates>
    <template name="test">
        <repeat _range="${r}">
          <set_variable name="${Ships.ship}" value="1.0"/>
        </repeat>
    </template>
  </templates>
  <values>
    <range name="Ships">
      <struct>
        <value ship="Artemis" />
        <range name="Fighters">
          <value pilot="Alta" hero="true" />
          <value pilot="Nimbus" hero="true" />
        </range>
      </struct>
      <struct>
        <value ship="Interpid" />
        <range name="Fighters">
          <value pilot="Remy" hero="true" />
          <value pilot="Jango" hero="true" />
        </range>
      </struct>

    </range>
  </values>

  <start>

    <repeat _range="Ships">
      <set_variable name="${Ships.ship}" value="1.0"/>

      <repeat _range="Fighters">
        <set_variable name="${Fighters.pilot}" value="2.0"/>
      </repeat>
    </repeat>
    <expand _template="test" r="Ships" />
      
    
  </start>


</mission_data>