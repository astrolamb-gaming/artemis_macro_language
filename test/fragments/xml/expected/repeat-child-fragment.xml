<mission_data version="1.0">
  <start>
    <set_variable name="Artemis" value="1.0"/>
    <set_variable name="Alta" value="2.0"/>
    <set_variable name="Nimbus" value="2.0"/>
    <set_variable name="Interpid" value="1.0"/>
    <set_variable name="Remy" value="2.0"/>
    <set_variable name="Jango" value="2.0"/>
    <set_variable name="Artemis" value="1.0"/>
    <set_variable name="Interpid" value="1.0"/>
    <set_variable name="Artemis" value="1.0"/>
    <set_variable name="Interpid" value="1.0"/>
  </start>
</mission_data>